rust/src/datatypes.rs

```diff
@@ -313,9 +301,13 @@ impl Field {
 
     // Get the max field id of itself and all children.
     fn max_id(&self) -> i32 {
-        min(
+        max(
             self.id,
-            self.children.iter().map(|c| c.id).min().unwrap_or(i32::MAX),
+            self.children
+                .iter()
+                .map(|c| c.max_id())
+                .max()
+                .unwrap_or(i32::MIN),
```

Comment:
nbd but i thought I saw elsewhere `-1` was the place holder? Is it too cumbersome to use `Option<u32>` as field/parent id?

Reply:
Oh, right, good catch. fixed.